syntax = "proto3";

package message;

message GetVerifyRequest {
  string email = 1;
}

message GetVerifyResponse {
  int32 error = 1;
  string email = 2;
  string code = 3;
}

service VerifyService {
  rpc GetVerifyCode (GetVerifyRequest) returns (GetVerifyResponse) {}
}

message GetChatServerRequest {
  int32 uid = 1;
}

message GetChatServerResponse {
  int32 error = 1;
  string host = 2;
  string port = 3;
  string token = 4;
}

message LoginRequest {
  int32 uid = 1;
  string token = 2;
}

message LoginResponse {
  int32 error = 1;
  int32 uid = 2;
  string token = 3;
}

service StatusService {
  rpc GetChatServer (GetChatServerRequest) returns (GetChatServerResponse) {}
  rpc Login (LoginRequest) returns (LoginResponse);
}
